packages/analytics-v1.1/src/core/analytics.js

```diff
@@ -603,30 +607,34 @@ class Analytics {
 
     // send the queued events to the fetched integration
     object.toBeProcessedByIntegrationArray.forEach(event => {
-      const methodName = event[0];
-      event.shift();
+      try {
```

Comment:
This is not necessary. We have to mainly target those code blocks where data is processed for destinations iteratively and those that can potentially prevent the events from going to the data plane.

Reply:
Let's audit the v3 code also to ensure they're isolated.

Reply:
Addressed

Reply:
We can add error handling in the event repository (v3) around the individual plugin invocations.

Reply:
Addressed